session/session.js

```diff
@@ -7,137 +7,168 @@
+  #req;
+
+  /**
+   * The request sessionID at the time of Session construction, i.e., req.sessionID.
+   * @type {string}
+   */
+  // @ts-ignore
+  #id;
+
+  /**
+   * Each session has a unique ID associated with it.
+   * This property is an alias of `req.sessionID` and cannot be modified.
+   * It has been added to make the session ID accessible from the session object.
+   * @returns {string}
+   * @public
+   */
+  get id() {
+    return this.#id
+  }
 
-  if (typeof data === 'object' && data !== null) {
-    // merge data into this, ignoring prototype properties
-    for (var prop in data) {
-      if (!(prop in this)) {
-        this[prop] = data[prop]
+  /**
+   * Create a new `Session` with the given request and `data`.
+   *
+   * @constructor
+   * @param {Request} req
+   * @param {Partial<SessionData>} data
+   */
+  constructor(req, data) {
+    this.#req = req
+    this.#id = req.sessionID
+
+    if (typeof data === 'object' && data !== null) {
+      /** @type {{[x: string]: any}} */
+      const thisAsObject = this
+      /** @type {{[x: string]: any}} */
+      const dataAsObject = data
+      // merge data into this, ignoring prototype properties
+      for (var prop in dataAsObject) {
+        if (!(prop in thisAsObject)) {
+          thisAsObject[prop] = dataAsObject[prop]
+        }
       }
     }
   }
-}
-
-/**
- * Update reset `.cookie.maxAge` to prevent
- * the cookie from expiring when the
- * session is still active.
- *
- * @return {Session} for chaining
- * @api public
- */
-
-defineMethod(Session.prototype, 'touch', function touch() {
-  return this.resetMaxAge();
-});
 
-/**
- * Reset `.maxAge` to `.originalMaxAge`.
- *
- * @return {Session} for chaining
- * @api public
- */
-
-defineMethod(Session.prototype, 'resetMaxAge', function resetMaxAge() {
-  this.cookie.maxAge = this.cookie.originalMaxAge;
-  return this;
-});
-
-/**
- * Save the session data with optional callback `fn(err)`.
- *
- * @param {Function} fn
- * @return {Session} for chaining
- * @api public
- */
-
-defineMethod(Session.prototype, 'save', function save(fn) {
-  this.req.sessionStore.set(this.id, this, fn || function(){});
-  return this;
-});
+  /**
+   * Update reset `.cookie.maxAge` to prevent
+   * the cookie from expiring when the
+   * session is still active.
+   *
+   * @return {Session} for chaining
+   * @public
+   */
+  touch() {
+    return this.resetMaxAge();
+  }
 
-/**
- * Re-loads the session data _without_ altering
- * the maxAge properties. Invokes the callback `fn(err)`,
- * after which time if no exception has occurred the
- * `req.session` property will be a new `Session` object,
- * although representing the same session.
- *
- * @param {Function} fn
- * @return {Session} for chaining
- * @api public
- */
+  /**
+   * Reset `.maxAge` to `.originalMaxAge`.
+   *
+   * @this {Session & Partial<SessionData>}
+   * @return {Session} for chaining
+   * @public
+   */
+  resetMaxAge() {
+    if (this.cookie) {
```

Comment:
This is a "fix` that is result of a type checking. The `Session.cookie` can be undefined. If I remember right, there is issue about some edge case when this happens.

At least, this happens when there is a code bug in `Session.constructor()`.